src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -182,3 +223,93 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         }
     }
 }
+
+/// The function `convert_record_batch_to_stream_chunk` is designed to transform the given `RecordBatch` into a `StreamChunk`.
+///
+/// For each column in the source column:
+/// - If the column's schema matches a column in the `RecordBatch` (both the data type and column name are the same),
+///   the corresponding records are converted into a column of the `StreamChunk`.
+/// - If the column's schema does not match, null values are inserted.
+/// - Hidden columns are handled separately by filling in the appropriate fields to ensure the data chunk maintains the correct format.
+/// - If a column in the Parquet file does not exist in the source schema, it is skipped.
+///
+/// # Arguments
+///
+/// * `record_batch` - The `RecordBatch` to be converted into a `StreamChunk`.
+/// * `source_columns` - User defined source schema.
+///
+/// # Returns
+///
+/// A `StreamChunk` containing the converted data from the `RecordBatch`.
+fn convert_record_batch_to_stream_chunk(
+    record_batch: RecordBatch,
+    source_columns: Option<Vec<Column>>,
+    file_name: String,
+) -> Result<StreamChunk, crate::error::ConnectorError> {
+    match source_columns {
+        Some(source_columns) => {
+            let size = source_columns.len();
+            let mut chunk_columns = Vec::with_capacity(source_columns.len() + 3);
```

Comment:
why +3, a little dangerous to hard code the length

Reply:
Oh, forgot to change this.

Reply:
Here I just want to give chunk_columns a pre-allocated capacity. The actual capacity is the length of source_columns plus the hidden columns. There are two hidden columns that can be confirmed here, file_name and offset. I am not sure whether there is a row_id. Since the final size is unknown here, if `source_columns.len()` is used, it will eventually become twice the capacity according to the Vec expansion rule.

For example, the length of the source column is 100, and there is a row_id column :
`let chunk_columns = Vec::with_capacity(source_columns.len() );` => Length: 103
Capacity: 200
`let chunk_columns = Vec::with_capacity(source_columns.len() +2);` => Length: 103
Capacity: 200
`let chunk_columns = Vec::with_capacity(source_columns.len() +3);` => Length: 103
Capacity: 103